crates/key-value-aws/src/store.rs

```diff
@@ -0,0 +1,517 @@
+impl ProvideCredentials for KeyValueAwsDynamoRuntimeConfigOptions {
+    fn provide_credentials<'a>(
+        &'a self,
+    ) -> aws_credential_types::provider::future::ProvideCredentials<'a>
+    where
+        Self: 'a,
+    {
+        aws_credential_types::provider::future::ProvideCredentials::ready(Ok(Credentials::new(
+            self.access_key.clone(),
+            self.secret_key.clone(),
+            self.token.clone(),
+            None, // Optional expiration time
+            "spin_custom_aws_provider",
+        )))
+    }
+}
+
+/// AWS Dynamo Key / Value enumeration for the possible authentication options
+#[derive(Clone, Debug)]
+pub enum KeyValueAwsDynamoAuthOptions {
+    /// Runtime Config values indicates credentials have been specified directly
+    RuntimeConfigValues(KeyValueAwsDynamoRuntimeConfigOptions),
+    /// Environmental indicates that the environment variables of the process should be used to
+    /// create the SDK Config for the Dynamo client. This will use the AWS Rust SDK's
+    /// aws_config::load_defaults to derive credentials based on what environment variables
+    /// have been set.
+    ///
+    /// See https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-authentication.html for options.
+    Environmental,
+}
+
+impl KeyValueAwsDynamo {
+    pub fn new(
+        region: String,
+        table: String,
+        auth_options: KeyValueAwsDynamoAuthOptions,
+    ) -> Result<Self> {
+        let region_clone = region.clone();
+        let client_fut = Box::pin(async move {
+            let sdk_config = match auth_options {
+                KeyValueAwsDynamoAuthOptions::RuntimeConfigValues(config) => SdkConfig::builder()
+                    .credentials_provider(SharedCredentialsProvider::new(config))
+                    .region(Region::new(region_clone))
+                    .behavior_version(BehaviorVersion::latest())
+                    .build(),
+                KeyValueAwsDynamoAuthOptions::Environmental => {
+                    aws_config::load_defaults(BehaviorVersion::latest()).await
+                }
+            };
+            Client::new(&sdk_config)
+        });
+
+        Ok(Self {
+            table: Arc::new(table),
+            region: Arc::new(region),
+            client: async_once_cell::Lazy::from_future(client_fut),
+        })
+    }
+}
+
+#[async_trait]
+impl StoreManager for KeyValueAwsDynamo {
+    async fn get(&self, name: &str) -> Result<Arc<dyn Store>, Error> {
+        Ok(Arc::new(AwsDynamoStore {
+            _name: name.to_owned(),
+            client: self.client.get_unpin().await.clone(),
+            table: self.table.clone(),
+        }))
+    }
+
+    fn is_defined(&self, _store_name: &str) -> bool {
+        true
+    }
+
+    fn summary(&self, _store_name: &str) -> Option<String> {
+        Some(format!(
+            "AWS DynamoDB region: {:?}, table: {}",
+            self.region, self.table
+        ))
+    }
+}
+
+struct AwsDynamoStore {
+    _name: String,
```

Comment:
Indeed, this appears to be dead -- further investigation shows it may also be dead in the Azure implementation (likely where I utilized my brilliant _ctrl+c, ctrl+v_ technique) -- I went ahead and removed this in both implementations